catalog/dags/providers/provider_api_scripts/europeana.py

```diff
@@ -223,7 +243,52 @@ def get_batch_data(self, response_json: dict) -> None | list[dict]:
         return response_json.get("items")
 
     def get_record_data(self, data: dict) -> dict:
-        return self.record_builder.get_record_data(data)
+        return self.record_builder.get_record_data(
+            data,
+            self._get_additional_item_data(data),
+        )
+
+    def _get_additional_item_data(self, data) -> dict:
+        # Delay of 30 seconds is fine for 100 items at a time, but not for each item.
+        # Occasionally getting this error on the item request when the delay is 3 secs:
+        # ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed
+        # connection without response'))
+        # But it always works on the next try so maybe that's ok?
+        # Options:
+        # - Switch to 3 second delay across the board and leave everything else alone,
+        #   or maybe add a max number of this kind of failure per batch or per dag run.
+        # - Separate delayed requesters 30 seconds for batch, 1 second for items, same
+        #   default batch size = 100.
+        # - Smaller batches (35), but default delay = 1 sec, like Brooklyn Museum.
+        # - Just go with defaults across the board, on the assumption that the batch
+        #   query is the expensive one, and there will be at least 30 seconds between,
+        #   given the 100 individual look-ups.
+        # Seems like might be best to reach out to Europeana and get their preference
+        # for the best way for us to proceed.
+        if not (item_id := data.get("id")):
+            logger.warning(f"No id, cannot request additional info on {data}")
+            return {}
+        item_response = self.get_response_json(
+            query_params=self.item_params,
+            endpoint=f"https://api.europeana.eu/record/v2{item_id}.json",
+        )
+        if not item_response or not item_response.get("success"):
+            logger.warning("Item request failed no response or ``success != True``")
+            return {}
+        # Assume that we just want the first info available in the item response,
+        # but consider adding checks, e.g. on landing url.
+        # Testing with the first webresource from the first aggregation, got dimensions
+        # for 1424 / 1589 images; so not too worried about performance implications of
+        # the loops. Limiting factor was much more the delay between requests.
+        if item_object := item_response.get("object"):
```

Comment:
We could probably reduce nesting if we used something like
`if aggregations := item_response.get("object", {}).get("aggregations"):` and 
`for webresource in aggregation.get("webResources", [])`